10-JavaScript/js-intro.md

```diff
@@ -0,0 +1,439 @@
+1. `apple;`
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
+
+  1. `Thrown: ReferenceError: apple is not defined`
+  1. `undefined`
+  1. `undefined`
+
+</details>
+
+<br/>
+
+What is our takeaway? When we work with variables in JavaScript, if we work with a variable with no defined value (as in, it's never been assigned a value with the assignment operator (`=`)), then we might see `undefined`.
+
+### Objects are Everything
+
+In JavaScript, [objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) are things that have name/value pairs.
+
+We will begin our understanding of objects in JavaScript by comparing them to hashes in Ruby. **However, it is crucial to know that JavaScript objects are very different from hashes in Ruby.** We will discover those differences later.
+
+Note the following similarities between JavaScript objects and Ruby hashes:
+
+- They are both declared (set-up) using curly braces (`{}`)
+- You can access properties with bracket notation
+- You can additionally access properties with dot notation
+
+In your Node REPL, copy and paste these lines. Do you get what you expect?
+
+1.
+    ```javascript
+    let testObj = {
+      someNum: 5,
+      someStr: 'this is a test string',
+      someNestedObj: {
+        someOtherNum: 4,
+      }
+    };
+    ```
+1. `testObj;`
+1. `testObj['someNum'];`
+1. `testObj.someNum;`
+1. `testObj.someStr;`
+1. `testObj.someNestedObj;`
+1. `testObj.someNestedObj.someOtherNum;`
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
+
+  1. `undefined`
+  1. 
+      ```javascript
+      { someNum: 5,
+        someStr: 'this is a test string',
+        someNestedObj: { someOtherNum: 4 } }
+      ```
+  1. `5`
+  1. `5`
+  1. `'this is a test string'`
+  1. `{ someOtherNum: 4 }`
+  1. `4`
+
+</details>
+
+### Functions
+
+We will spend the rest of our time in JavaScript thinking about functions, so stay tuned. ;)
+
+## What Did We Accomplish?
+
+- Install Node, so we can run JavaScript from the command line
+- Practice using the node REPL accessed with `$ node`
+- Compare the different data types between Ruby and JavaScript
+  - `Boolean`
+  - `Number`
+  - `String`
+  - `Array`
+  - `null`
+  - `undefined`
+  - `Object`
+  - `Function`
+
+## Summary: The Data Type Ruby vs. JavaScript Abbreviated List
+
+| Data Type | Things to note about Ruby vs. JavaScript |
+| --- | --- |
+| Boolean | -
+| Number | JavaScript has its own definitions for float math, infinity, and `NaN` (Not a Number)
+| String | We prefer making Strings with single-quotes (`''`). To do string interpolation, we need to use the syntax `${}` and backticks instead of single-quotes
+| Array | -
+| null | -
+| undefined | Will be useful when we think about variables that may have never been assigned a value
+| Object | Our base understanding will come from Ruby's hashes, but will soon evolve
+| Function | We will get into this more in-depth later
+
```

Comment:
Do we want to mention that Symbols aren't a thing in JS?

Reply:
nah, I thought about it, but it's not useful imo, it's just weird stuff used for JS internals. Until there starts to be a good use case for it, I'm fine with obscuring it. That's my current thought. Would you make the argument for symbols?